snooty/parser.py

```diff
@@ -102,6 +103,10 @@
 logger = logging.getLogger(__name__)
 
 
+class ProjectLoadError(Exception):
+    pass
```

Comment:
for my own understanding, exceptions will stop the process, as diagnostics will be appended to the log?

Reply:
Yep! This is raised *after* reporting diagnostics to the backend. So `main()` just catches this exception, flushes the backend so those diagnostics get reported, and then exits.

The second thing there is important: if the exception isn't caught, no diagnostics are reported. You just get an ugly backtrace.

Reply:
perfectimo! 👍